Library spring jpa

kondury opened the pull request:
hw 6 library spring jpa, dao and service tests

kondury committed:
hw5 library spring jdbc liquibase h2

kondury committed:
hw 6 library spring jpa, dao and service tests

stvort commented on library/library-spring-jpa/src/main/java/com/github/kondury/library/dao/JpaCommentDao.java:
```diff
@@ -0,0 +1,58 @@
+        return em.createQuery(
+                        """
+                                select comment from Comment comment
+                                join fetch comment.book
```
В данном случае, будет экономнее достать книгу отдельным запросом, нежели присоединять ее ко всем своим комментариям

stvort commented on library/library-spring-jpa/src/test/java/com/github/kondury/library/dao/JpaBookDaoTest.java:
```diff
@@ -0,0 +1,137 @@
+class JpaBookDaoTest {
+
+    @Autowired
+    BookDao bookDao;
```
`private`

kondury committed:
hw 5 made changes according a review
- fixed the visibility of injected fields to private in test classes
- fixed findByBookId() in JpaCommentDao class: replaced join fetch in select with preceding em.find()
